src/benchmark_environments/risky_path.py

```diff
@@ -0,0 +1,160 @@
+import gym
+from gym.spaces import MultiDiscrete
+
+import numpy as np
+
+import random
+import sys
+import time
+
+def _clamp(val, min_val, max_val):
+    if val < min_val:
+        return min_val
+    elif max_val < val:
+        return max_val
+    else:
+        return val
+
+class RiskyPathEnv(gym.Env):
+    """Gridworld designed to test basic understanding of causality under stochasticity.
+
+    This environment is proposed in section 6.3 of B. Ziebart's PhD thesis (2010).
+
+    The objective is to start in the top left corner and reach the lower bottom corner:
+
+    START
+     |
+     v
+     -.....
+     -****.
+     ----*.
+     ----*.
+     ----*.
+     -----_ <--GOAL
+
+    The actions are moving RIGHT, UP, LEFT and DOWN, and all tiles are walkable.
+    The reward is given by your current tile, as follows:
+
+    '*' : -100
+    '-' : -10
+    '.' : -1
+    '_' : 0
+
+    If the environment is deterministic, then the best strategy is to go through the
+    narrow path in the top right region; however, if we have noise in the movements
+    (i.e. 50% you move the direction you chose, %50 chance you choose an adjacent
+    direction), there is a chance of hitting the high cost region - because of this,
+    the best strategy is then to go through the left bottom region.
+    An agent that doesn't take into account the fact that it cannot choose its own
+    trajectory might try to take the riskier path. 
+
+    This is set up with a fixed episode length of 50, but movement stops once you
+    reach the goal.
+    """
+
+    def __init__(self):
+        self._width = 6
+        self._height = 6
+
+        self.observation_space = MultiDiscrete((self._height, self._width))
+        self.action_space = Discrete(4)
+
+        self._pos = None
+        self.max_episode_steps = 50
```

Comment:
Is there a reason we want to use this rather than `TimeLimitWrapper`?

Some reasons could be if we need to track time anyway (e.g. to include it in the state), or if we were doing some potential shaping where we need to change the reward for the terminal state. But I don't think either apply in the current implementation.